### PR TITLE
feat: add input bun-version-file

### PR DESCRIPTION
**Pull request by adeherysh**
## What does this PR do?
- feat: add input for bun-version-file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,12 +42,13 @@ In most cases, you shouldn't need to use the [setup-node](https://github.com/act
...
-| -------------- | -------------------------------------------------- | ----------- | ------------------------------- |
-| `bun-version`  | The version of Bun to download and install.        | `latest`    | `canary`, `1.0.0`, `1.0.x`      |
-| `registry-url` | Registry URL where some private package is stored. | `undefined` | `"https://npm.pkg.github.com/"` |
-| `scope`        | Scope for private packages.                        | `undefined` | `"@foo"`, `"@orgname"`          |
-| `no-cache`     | Disable caching of the downloaded executable.      | `false`     | `true`, `false`                 |
+| Name                | Description                                            | Default     | Examples                        |
+| ------------------- | ------------------------------------------------------ | ----------- | ------------------------------- |
+| `bun-version`       | The version of Bun to download and install.            | `latest`    | `canary`, `1.0.0`, `1.0.x`      |
+| `bun-version-file`  | The version of Bun to download and install from file.  | `undefined` | `canary`, `1.0.0`, `1.0.x`      |
```
Comment (xhyrom) [inline README.md]: Wrong examples for `bun-version-file`

**Review by xHyroM — CHANGES_REQUESTED** [review]

**Comment by xhyrom** [comment]
I'm not a big fan of `x-file` - it looks horrible. What about allowing files directly in `bun-version` instead of having two properties for the same thing?
Also this is a breaking change thus it would appear in `2.0.0`.

**Comment by adeherysh** [comment]
> I'm not a big fan of `x-file` - it looks horrible. What about allowing files directly in `bun-version` instead of having two properties for the same thing? Also this is a breaking change thus it would appear in `2.0.0`.

Imo this config very useful for monorepo when using single workflow and have diferrent bun version between package, just create .bumrc file inside each package. And this config can make consistency bun installed version between when running in local or ci

File: README.md (hunk 1 above)
Comment (adeherysh) [reply README.md]: Thanks, lemme check

**Review by xHyroM — APPROVED** [review]
LGTM; I'll create branch called `dev` because I want to do some more changes

**Comment by adeherysh** [comment]
> LGTM; I'll create branch called `dev` because I want to do some more changes

oke thanks @xHyroM

**Comment by michaelmior** [comment]
Is this supposed to be called `.bumrc` by default and not `.bunrc`?

**Comment by Jarred-Sumner** [comment]
> Is this supposed to be called `.bumrc` by default and not `.bunrc`?

It is not supposed to be called `.bumrc` or `.bunrc`

**Comment by michaelmior** [comment]
@Jarred-Sumner The example given is `.bumrc`, so I'm confused by your comment. I was just wondering if there was a typo that should be corrected.

**Comment by Jarred-Sumner** [comment]
Yep, thanks for flagging

**Comment by xhyrom** [comment]
> @Jarred-Sumner The example given is `.bumrc`, so I'm confused by your comment. I was just wondering if there was a typo that should be corrected.

Fixed in https://github.com/oven-sh/setup-bun/pull/86, thanks for flagging :)

**Pull request merged**